projects/clr/hipamd/src/hip_device.cpp

```diff
@@ -766,13 +766,16 @@ hipError_t hipGetDevicePropertiesR0000(hipDeviceProp_tR0000* prop, int device) {
   HIP_RETURN(hipSuccess);
 }
 
-hipError_t hipGetProcAddress(const char* symbol, void** pfn, int hipVersion, uint64_t flags,
+hipError_t hipGetProcAddress_common(const char* symbol, void** pfn, int hipVersion, uint64_t flags,
                              hipDriverProcAddressQueryResult* symbolStatus) {
-  HIP_INIT_API(hipGetProcAddress, symbol, pfn, hipVersion, flags, symbolStatus);
-
   std::string symbolString = symbol;
-  if (symbol == nullptr || symbolString == "" || pfn == nullptr) {
-    HIP_RETURN(hipErrorInvalidValue);
+  if(symbol == nullptr || symbolString == "" || pfn == nullptr) {
```

Comment:
```suggestion
  if(symbol == nullptr || std::string_view{symbol}.empty() || pfn == nullptr) {
```
constructing a std::string from a nullptr is undefined behavior.

Reply:
also, the actual construction line (i.e. `std::string symbolString = symbol;`) needs to be removed or gated as well.
Some implementations will throw an exception when constructing from a nullptr.